pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -121,8 +123,16 @@ type NodeInfo struct {
 }
 
 // New creates an instance of a mon cluster
-func New(context *clusterd.Context, namespace, dataDirHostPath, rookVersion string, cephVersion cephv1.CephVersionSpec, mon cephv1.MonSpec,
-	placement rookalpha.Placement, hostNetwork bool, resources v1.ResourceRequirements, ownerRef metav1.OwnerReference) *Cluster {
+func New(
```

Comment:
As another general statement, it seems like every daemon has the same kind of struct in their setup, and they all have things like context, namespace, cephVersison, hostNetwork, etc. I would propose to create a new context struct for ceph that carries with it the clusterd.Context and the cluster CRD info that can be the same for all daemons. This struct would be created by the operator and passed to every daemon operator process.